socket_wrapper.cpp

```diff
@@ -30,43 +30,60 @@ void SocketWrapper::InitSocketAddress(int family, int address, int port)
     m_socket_address.sin_port = port;
 }
 
-void SocketWrapper::SetInAddressWithStr(const char* ip_address_str)
+void Socket::SetInAddressWithStr(const char* ip_address_str)
 {
     if((inet_pton(AF_INET, ip_address_str, &m_socket_address.sin_addr)) <= 0)
-        throw socket_exception("inet_pton returned error");
+        throw SocketException("inet_pton returned error");
 }
 
-void SocketWrapper::Bind()
+void Socket::Bind(const unsigned address, const unsigned port)
 {
+    m_socket_address.sin_addr.s_addr = address;
+    m_socket_address.sin_port = port;
+
     if(bind(m_socket_handle, (const sockaddr*)&m_socket_address, sizeof(m_socket_address)) < 0)
-        throw socket_exception("bind returned error");
+        throw SocketException("bind returned error");
 }
 
-void SocketWrapper::Listen(int flag)
+void Socket::Listen(int flag)
```

Comment:
Seems like the parameter name `flag` is not what this param really is. Lets choose some more consistent name for it (see docs for `listen` function).